presto-native-execution/presto_cpp/main/Announcer.cpp

```diff
@@ -98,106 +100,10 @@ Announcer::Announcer(
           nodeLocation,
           connectorIds)),
       announcementRequest_(
-          announcementRequest(address, port, nodeId, announcementBody_)),
-      pool_(velox::memory::addDefaultLeafMemoryPool("Announcer")),
-      eventBaseThread_(false /*autostart*/),
-      clientCertAndKeyPath_(clientCertAndKeyPath),
-      ciphers_(ciphers) {}
-
-void Announcer::start() {
-  eventBaseThread_.start("Announcer");
-  stopped_ = false;
-  auto* eventBase = eventBaseThread_.getEventBase();
-  eventBase->runOnDestruction([this] { client_.reset(); });
-  eventBase->schedule([this]() { return makeAnnouncement(); });
-}
-
-void Announcer::stop() {
-  stopped_ = true;
-  eventBaseThread_.stop();
-}
-
-void Announcer::makeAnnouncement() {
-  // stop() calls EventBase's destructor which executed all pending callbacks;
-  // make sure not to do anything if that's the case
-  if (stopped_) {
-    return;
-  }
-
-  try {
-    auto newAddress = coordinatorDiscoverer_->updateAddress();
-    if (newAddress != address_) {
-      LOG(INFO) << "Discovery service changed to " << newAddress.getAddressStr()
-                << ":" << newAddress.getPort();
-      std::swap(address_, newAddress);
-      client_ = std::make_shared<http::HttpClient>(
-          eventBaseThread_.getEventBase(),
-          address_,
-          std::chrono::milliseconds(10'000),
-          pool_,
-          clientCertAndKeyPath_,
-          ciphers_);
-    }
-  } catch (const std::exception& ex) {
-    LOG(WARNING) << "Error occurred during announcement run: " << ex.what();
-    scheduleNext();
-    return;
-  }
-
-  client_->sendRequest(announcementRequest_, announcementBody_)
-      .via(eventBaseThread_.getEventBase())
-      .thenValue([this](auto response) {
-        auto message = response->headers();
-        if (message->getStatusCode() != http::kHttpAccepted) {
```

Comment:
CC: @xiaoxmeng 
Original announcement is only checking status code, not content empty